tests/tests/mmap.sh

```diff
@@ -0,0 +1,186 @@
+#
+# test mmap() and normal read/write consistency between different nodes
+#
+# This script opens a file with a fixed size and writes a pattern
+# at random offset with random length, and then reads it again.
```

Comment:
Ah.  I had pictured a simpler ping pong unit test to just make sure that the basics worked.  If we're getting into random (ish) fuzz testing then we should use fsx-mpi.  It's a bit like this, but it races the operations, can do more (truncate), and stops and gives you a dump of the problem and operations that lead to it when it sees a problem.

Reply:
I've repaced it with a C program which is simpler and slightly faster than doing this using xfs_io and scripts, and as a bonus it just works on el7 as well. We can do fsx-mpi stuff maybe later, I'm thinking. For now this does a good job at hitting the lock contention.